chain_bridge.go

```diff
@@ -81,6 +81,38 @@ func (l *LndRpcChainBridge) GetBlockHash(ctx context.Context,
 	return blockHash, nil
 }
 
+// VerifyBlock returns an error if a block (with given header and height) is not
+// present on-chain. It also checks to ensure that block height corresponds to
+// the given block header.
+func (l *LndRpcChainBridge) VerifyBlock(ctx context.Context,
+	header wire.BlockHeader, height uint32) error {
+
+	_, err := l.GetBlock(
```

Comment:
I think we should do this last, only once we know we're expecting the block to be in the current chain (since this is the most "expensive" call in terms of resources).